### PR TITLE
Add performance writeup to ClimaCore

### PR DESCRIPTION
**Pull request by charleskawczynski**
I wasn't sure exactly where to put this performance writeup, is this okay? Or should I transfer to overleaf?

### CODE REVIEW

**Comment by valeriabarra** [comment]
Hi Charlie, thanks for this write up! This is great! Do you think we could use some of this material for a workshop? In that case, I would think we could later port some of it to the CliMAWorkshop repo, together with a non-optimized example to show how to apply these recommendations. 

Let me know what you think. 
Thanks!

**Comment by charleskawczynski** [comment]
> Hi Charlie, thanks for this write up! ...

I'm not sure if I'll have the bandwidth for the CliMA workshop, let's play it by ear. I'm fine if people would like to port this to the workshop repo, though.

**Review by simonbyrne — APPROVED** [review]

File: docs/src/performance_tips.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+CliMA's [ReportMetrics.jl](https://github.com/CliMA/ReportMetrics.jl) applies the strategy in the above section and provides a re-useable interface for reporting the top-most important allocations. Here is an example of it in use:
+
+ - rep_workload.jl
+ - perf.jl
+
+```julia
+# File: rep_workload.jl
+import Profile
+
+import SpecialFunctions # for triggering invalidations
```
Comment (valeriabarra) [inline docs/src/performance_tips.md:50]: Thanks Charlie for adding this! 
Just a minor question. What do you need this package for here?

Comment (charleskawczynski) [reply docs/src/performance_tips.md:50]: Ah, yeah, I can remove this, it was for something else

Comment (valeriabarra) [reply docs/src/performance_tips.md:50]: That'd be nice. Thanks!

**Review by valeriabarra — APPROVED** [review]

**Pull request merged**